### ISSUE TITLE
Error connecting to the service protocol: HttpException: Connection closed before full header was received

### ISSUE THREAD

**Issue by mpannu03** [Lyokone/flutterlocation#154]
When I reverted to 2.3.0 everything became fine.

**Comment by Lyokone** [comment]
Hey :)
My plugin does not use HTTP connection. Can you give me the part of the code where the error is happening ?

**Comment by rIskJunkie** [comment]
Have the same issue.
Not clear which part of code is doing this.
Will try to revert to 2.3.0 and see.

**Comment by Lyokone** [comment]
Hey, can you try again with 2.3.4 and let me know if anything change ? Do you use any other plugins ? On which platform are you having this error ?

**Comment by burntblark** [comment]
Hi @Lyokone. Thanks for the plugin BTW. However, I am facing the same issue here with the current version (2.3.5). Without your plugin, everything works fine.

Other plugins I am using?
`google_maps_flutter: ^0.5.5`

Platform?
Android 9.0 - Emulator (Pixel API 28)

**Comment by charzone95** [comment]
Hi @Lyokone 
Just encountered the same problem here.

`Error connecting to the service protocol: HttpException: Connection closed before full header was received, uri = http://127.0.0.1:23845/ws`

Using API Pixel 2 API 28 (Pie)

Dependencies:
dependencies:
```
  flutter:
    sdk: flutter

  cupertino_icons: ^0.1.2
  google_maps_flutter: ^0.5.7
  location: ^2.3.5
```

**Comment by toast-uz** [comment]
I've had the same issue.

IDE: Android Studio 3.4
Emulator: PIXEL2 API28, Android9.0

First, created New Flutter Application Project, with no sample content. It ran normally as "Flutter Demo Home Page".

Then, only added dependencies "location: ^2.3.5" on pubspec.yaml. It could not run normally. I did not touch anything else.

```
Running "flutter packages get" in flutter_app...                   1.6s
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
Running Gradle task 'assembleDebug'...
Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...
Error connecting to the service protocol: HttpException: Connection closed before full header was received, uri = http://127.0.0.1:57484/ws
```

**Comment by toast-uz** [comment]
Tried on some environments:

location: ^2.3.5, PIXEL2 API27 (Android8.1) --> same issue
location: ^2.3.5, Nexus5X API24 (Android7.0) --> same issue
location: ^2.3.5, iPhone Xr 12.2 --> no issue, run normally
delete dependencies, PIXEL2 API28 (Android9.0) --> no issue, run normally
location: ^2.3.0, PIXEL2 API28 (Android9.0) --> same issue
location: ^2.2.0, PIXEL2 API28 (Android9.0) --> same issue
location: ^2.1.0, PIXEL2 API28 (Android9.0) --> same issue
location: ^2.0.0, PIXEL2 API28 (Android9.0) --> same issue
location: ^1.4.0, PIXEL2 API28 (Android9.0) --> no issue, run normally

The issue seems to be implemented at 2.0.0, and reproduced only on Android.

**Comment by toast-uz** [comment]
Sorry, I've resolved by upgrading my gradle as README.md .  Thank you!
